app/src/main/java/io/apicurio/registry/rest/RulesResourceImpl.java

```diff
@@ -19,74 +19,131 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
 import io.apicurio.registry.rest.beans.Rule;
+import io.apicurio.registry.storage.RegistryStorage;
+import io.apicurio.registry.storage.RegistryStorageException;
+import io.apicurio.registry.storage.RuleAlreadyExistsException;
+import io.apicurio.registry.storage.RuleConfigurationDto;
+import io.apicurio.registry.storage.RuleNotFoundException;
 
 /**
  * Implementation of the @RulesResource JAX-RS interface.
  * @author eric.wittmann@gmail.com
  */
+@ApplicationScoped
 public class RulesResourceImpl implements RulesResource {
 
+    @Inject
+    RegistryStorage storage;
+    @Inject
+    ErrorFactory errorFactory;
+
     /**
      * @see io.apicurio.registry.rest.RulesResource#listGlobalRules()
      */
     @Override
     public List<Rule> listGlobalRules() {
-        // TODO provide a real implementation of this.
-        List<Rule> dummyData = new ArrayList<Rule>();
-        
-        Rule dummyRule = new Rule();
-        dummyRule.setName("DummyRule");
-        dummyRule.setConfig("FULL");
-        
-        dummyData.add(dummyRule);
-        
-        return dummyData;
+        try {
+            List<String> names = storage.getGlobalRules();
+            List<Rule> rules = new ArrayList<>();
+            for (String ruleName : names) {
+                try {
+                    RuleConfigurationDto ruleDto = storage.getGlobalRule(ruleName);
+                    Rule rule = new Rule();
+                    rule.setName(ruleName);
+                    // TODO set the configuration data on the rule
+                } catch (RuleNotFoundException rnfe) {
```

Comment:
Just return a list of dtos, hence no need to handle race condition?

Reply:
Good point, although I think it might be better to change the REST API to just return the rule names.  I'll do that and we can switch back if needed.